### PR TITLE
Expand fusion test coverage

### PR DESCRIPTION
**Pull request by ahojnnes**
## Summary
- Expand test coverage for `mvs/fusion.cc` (81.1% coverage, 49 uncovered lines)
- Add tests for previously uncovered fusion pipeline edge cases
- Targets item #14 from test coverage opportunities

## Test plan
- [ ] CI passes all existing and new tests
- [ ] New tests build and run successfully

### CODE REVIEW

File: src/colmap/mvs/fusion_test.cc [hunk 1]
```diff
@@ -139,6 +157,440 @@ TEST(StereoFusion, Integration) {
...
+  options.mask_path = mask_dir;
+
+  StereoFusion fusion(options, ws.temp_dir, "COLMAP", "", "geometric");
+  fusion.Run();
+
+  // All pixels masked out, so no points should be fused.
+  EXPECT_EQ(fusion.GetFusedPoints().size(), 0);
+}
+
+TEST(StereoFusion, PartialMaskReducesPoints) {
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/mvs/fusion_test.cc:471]: Keeping tests self-contained for clarity.

File: src/colmap/mvs/fusion_test.cc [hunk 2]
```diff
@@ -139,6 +157,440 @@ TEST(StereoFusion, Integration) {
...
+  options.min_num_pixels = 2;
+
+  StereoFusion fusion(options, ws.temp_dir, "COLMAP", "", "geometric");
+  fusion.Run();
+
+  // With perfectly aligned normals (all pointing in z direction), some points
+  // may still pass. The key test is that the code path for normal checking
+  // is exercised without crashing.
+  const auto& points = fusion.GetFusedPoints();
+  EXPECT_EQ(points.size(), fusion.GetFusedPointsVisibility().size());
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/mvs/fusion_test.cc:588]: Keeping tests self-contained for clarity.
